compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -1155,7 +1156,8 @@ public void visit(BLangFail failNode) {
     @Override
     public void visit(BLangSimpleVariableDef astVarDefStmt) {
         VarKind kind;
-        if (astVarDefStmt.var.symbol.origin == SymbolOrigin.VIRTUAL) {
+        if (astVarDefStmt.var.symbol.origin == SymbolOrigin.VIRTUAL ||
```

Comment:
Shall we use separate VarKind for let expression? Should we sepearate out expression variable in bir @manuranga

Reply:
I don't this is right. `SYNTHETIC` kind is for variables that user didn't write themself. But if you look the original issue, both `hello` and `ballerina` are coming from the source, not desugered.  So why should they be marked `SYNTHETIC`.

Reply:
We need to have a way to separately identify let expression inner variables among the function local variables. Currently let expression inner variables also marked with the variable kind `LOCAL` and we can't differentiate them from other local variables to separately handle their visibility scopes. Yes `SYNTHETIC` kind has been introduced for the variables generated at desugar (usually these variable names starts with specific symbols like $). So, using `SYNTHETIC` kind for  let expression inner variable might not be a good approach. Is it okay to add a new variable kind for expression inner variables?